<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="description" content="Package documentation for Agony" /><meta name="author" content="Jermiah Joseph" />
      <link rel="shortcut icon" href="img/favicon.ico" />
    <title>Agony</title>
    <link rel="stylesheet" href="css/theme.css" />
    <link rel="stylesheet" href="css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Home";
        var mkdocs_page_input_path = "index.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="." class="icon icon-home"> Agony
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="./search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
                <li class="toctree-l1 current"><a class="reference internal current" href="#">Home</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#introduction">Introduction</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#set-up">Set Up</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#prerequisites">Prerequisites</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#installation">Installation</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#usage">Usage</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#documentation">Documentation</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#project-ideas-and-next-steps">Project Ideas and Next Steps</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#project-ideas">Project Ideas</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#references">References</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#annotated-list-of-papers">ðŸ“„ Annotated List of Papers</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#1-inferring-genome-wide-interaction-networks-using-the-phi-mixing-coefficient">[1.] Inferring Genome-Wide Interaction Networks Using the Phi-Mixing Coefficient</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#2-finding-hierarchy-in-directed-online-social-networks">[2.] Finding Hierarchy in Directed Online Social Networks</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#3-faster-way-to-agony-discovering-hierarchies-in-directed-graphs">[3.] Faster Way to Agony: Discovering Hierarchies in Directed Graphs</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#4-tiers-for-peers-a-practical-algorithm-for-discovering-hierarchy-in-weighted-networks">[4.] Tiers for Peers: A Practical Algorithm for Discovering Hierarchy in Weighted Networks</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#5-reconstructing-directed-gene-regulatory-network-by-only-gene-expression-data">5. Reconstructing Directed Gene Regulatory Network by Only Gene Expression Data</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#6-resolution-of-ranking-hierarchies-in-directed-networks">6. Resolution of Ranking Hierarchies in Directed Networks</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#7-scenic-single-cell-regulatory-network-inference-and-clustering">7. SCENIC: single-cell regulatory network inference and clustering</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#8-inferring-regulatory-networks-from-expression-data-using-tree-based-methods">8. Inferring Regulatory Networks from Expression Data Using Tree-Based Methods</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#9-blars-an-algorithm-to-infer-gene-regulatory-networks">9. bLARS: An Algorithm to Infer Gene Regulatory Networks</a>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    </ul>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="usage/">Usage</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="data_sources/">Data Sources</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="devnotes/">Developer Notes</a>
                </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href=".">Agony</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="." class="icon icon-home" aria-label="Docs"></a></li>
      <li class="breadcrumb-item active">Home</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/bhklab/agony/edit/master/docs/index.md">Edit on agony</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="agony">Agony</h1>
<h2 id="introduction">Introduction</h2>
<p>The notion of <strong>graph agony</strong> was developed out of a desire to find hierarchies in directed networks <a href="#2">[2]</a>. Intuitively, a hierarchy -- in an organization or social network, for example -- involves assigning <em>levels</em> or <em>ranks</em> to people such that the ranks are ordered. For example, the CEO of a company should rank higher than an intern. </p>
<p>Agony works by trying to <em>find</em> a hierarchy that best reflects that structure of a given directed network. Specifically we set in</p>
<p>$$a$$</p>
<p><strong>Authors:</strong> <a href="https://github.com/jbannon">James Bannon</a>, <a href="https://github.com/julianguyn">Julia Nguyen</a>, <a href="https://github.com/mattbocc">Matthew Boccalon</a>, <a href="https://github.com/jjjermiah">Jermiah Joseph</a></p>
<p><strong>Contact:</strong> <a href="mailto:bhklab.jermiahjoseph@gmail.com">bhklab.jermiahjoseph@gmail.com</a></p>
<p><strong>Description:</strong> Agony implementation</p>
<hr />
<p><a href="https://github.com/prefix-dev/pixi"><img alt="pixi-badge" src="https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/prefix-dev/pixi/main/assets/badge/v0.json&amp;style=flat-square" /></a>
<a href="https://github.com/astral-sh/ruff"><img alt="Ruff" src="https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/astral-sh/ruff/main/assets/badge/v2.json&amp;style=flat-square" /></a>
<a href="https://github.com/squidfunk/mkdocs-material"><img alt="Built with Material for MkDocs" src="https://img.shields.io/badge/mkdocs--material-gray?logo=materialformkdocs&amp;style=flat-square" /></a></p>
<p><img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/bhklab/agony?style=flat-square" />
<img alt="GitHub issues" src="https://img.shields.io/github/issues/bhklab/agony?style=flat-square" />
<img alt="GitHub pull requests" src="https://img.shields.io/github/issues-pr/bhklab/agony?style=flat-square" />
<img alt="GitHub contributors" src="https://img.shields.io/github/contributors/bhklab/agony?style=flat-square" />
<img alt="GitHub release (latest by date)" src="https://img.shields.io/github/v/release/bhklab/agony?style=flat-square" /></p>
<h2 id="set-up">Set Up</h2>
<h3 id="prerequisites">Prerequisites</h3>
<p>Pixi is required to run this project.
If you haven't installed it yet, <a href="https://pixi.sh/latest/">follow these instructions</a></p>
<h3 id="installation">Installation</h3>
<ol>
<li>Clone this repository to your local machine</li>
<li>Navigate to the project directory</li>
<li>Set up the environment using Pixi:</li>
</ol>
<pre><code class="language-bash">pixi install
</code></pre>
<h3 id="usage">Usage</h3>
<p>agony bindings are available in the <code>agony</code> module.</p>
<p>I've setup two tasks to showcase this</p>
<pre><code class="language-console">pixi run help
</code></pre>
<pre><code class="language-console">pixi run cycle_dfs
</code></pre>
<h2 id="documentation">Documentation</h2>
<p>Click <a href="https://bhklab.github.io/agony">here</a> to view the full documentation.</p>
<h2 id="project-ideas-and-next-steps">Project Ideas and Next Steps</h2>
<h3 id="project-ideas">Project Ideas</h3>
<h2 id="references">References</h2>
<h3 id="annotated-list-of-papers">ðŸ“„ Annotated List of Papers</h3>
<hr />
<h4 id="1-inferring-genome-wide-interaction-networks-using-the-phi-mixing-coefficient"><a id="1">[1.]</a> <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC7731978/pdf/nihms-1539098.pdf">Inferring Genome-Wide Interaction Networks Using the Phi-Mixing Coefficient</a></h4>
<p>^[This paper introduces a novel algorithm based on the Ï†-mixing coefficient from probability theory to infer genome-wide interaction networks. The method allows for the construction of weighted and directed networks that can contain cycles, and it has been applied to study subtypes of lung cancer, including small cell (SCLC) and non-small cell (NSCLC), as well as normal lung tissue. There is a <a href="https://github.com/nitinksingh/phixer/tree/master">github repo</a> with an implementation that is a combination of C and Matlab.]</p>
<hr />
<h4 id="2-finding-hierarchy-in-directed-online-social-networks"><a id = "2">[2.]</a>  <a href="https://dl.acm.org/doi/10.1145/1963405.1963484">Finding Hierarchy in Directed Online Social Networks</a></h4>
<p>^<a href="../{&quot;attribution&quot;:{&quot;attributableIndex&quot;:&quot;717-1&quot;}}">The authors define a measure of hierarchy within directed online social networks and present an efficient algorithm to compute this measure. This work provides insights into the structural organization of social networks and how hierarchical relationships can be identified and quantified.</a> <a href="https://dl.acm.org/doi/10.1145/1963405.1963484?utm_source=chatgpt.com">oai_citation:0â€¡ACM Digital Library</a></p>
<hr />
<h4 id="3-faster-way-to-agony-discovering-hierarchies-in-directed-graphs"><a id="3">[3.]</a> <a href="https://arxiv.org/pdf/1902.01477">Faster Way to Agony: Discovering Hierarchies in Directed Graphs</a></h4>
<p>^<a href="../{&quot;attribution&quot;:{&quot;attributableIndex&quot;:&quot;1127-1&quot;}}">This paper presents an improved algorithm for computing "agony," a metric used to quantify the hierarchical structure in directed graphs. The proposed method reduces the computational complexity from O(nmÂ²) to O(mÂ²), making it more practical for analyzing large-scale networks.</a> <a href="https://arxiv.org/abs/1902.01477?utm_source=chatgpt.com">oai_citation:1â€¡arXiv</a></p>
<hr />
<h4 id="4-tiers-for-peers-a-practical-algorithm-for-discovering-hierarchy-in-weighted-networks"><a id="4">[4.]</a> <a href="https://arxiv.org/pdf/1903.02999">Tiers for Peers: A Practical Algorithm for Discovering Hierarchy in Weighted Networks</a></h4>
<p>^<a href="../{&quot;attribution&quot;:{&quot;attributableIndex&quot;:&quot;1522-1&quot;}}">Building upon the concept of agony, this study extends the approach to weighted networks and introduces constraints on the number of hierarchical levels. The authors connect the problem to the capacitated circulation problem and provide both exact and heuristic solutions, demonstrating efficiency in handling large datasets.</a></p>
<hr />
<h4 id="5-reconstructing-directed-gene-regulatory-network-by-only-gene-expression-data">5. <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5001240/">Reconstructing Directed Gene Regulatory Network by Only Gene Expression Data</a></h4>
<p>^<a href="../{&quot;attribution&quot;:{&quot;attributableIndex&quot;:&quot;1987-1&quot;}}">The paper proposes the Context-Based Dependency Network (CBDN) method to reconstruct directed gene regulatory networks using solely gene expression data. This approach addresses the challenge of inferring regulatory directions without additional data, such as eQTLs or gene knock-out experiments, which are often unavailable for human tissue samples.</a></p>
<hr />
<h4 id="6-resolution-of-ranking-hierarchies-in-directed-networks">6. <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0191604">Resolution of Ranking Hierarchies in Directed Networks</a></h4>
<p>^[ranked stochastic block models]</p>
<hr />
<h4 id="7-scenic-single-cell-regulatory-network-inference-and-clustering">7. <a href="https://www.nature.com/articles/nmeth.4463">SCENIC: single-cell regulatory network inference and clustering</a></h4>
<p>^[A paper that infers single cell gene regulatory networks. ]</p>
<hr />
<h4 id="8-inferring-regulatory-networks-from-expression-data-using-tree-based-methods">8. <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0012776">Inferring Regulatory Networks from Expression Data Using Tree-Based Methods</a></h4>
<p>^[The GENIE3 algorithm for inferring directed gene regulatory networks. There exists a <a href="https://github.com/vahuynh/GENIE3">github repo</a> with implementations in C, R, Matlab, and python. This is used in the SCENIC pipeline.]</p>
<hr />
<h4 id="9-blars-an-algorithm-to-infer-gene-regulatory-networks">9. <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&amp;arnumber=7138615">bLARS: An Algorithm to Infer Gene Regulatory Networks</a></h4>
<p>^[Another algorithm for inferring directed gene regulatory networks from gene expression.]</p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="usage/" class="btn btn-neutral float-right" title="Usage">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/bhklab/agony" class="fa fa-code-fork" style="color: #fcfcfc"> agony</a>
        </span>
    
    
    
      <span><a href="usage/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="js/jquery-3.6.0.min.js"></script>
    <script>var base_url = ".";</script>
    <script src="js/theme_extra.js"></script>
    <script src="js/theme.js"></script>
      <script src="search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>

<!--
MkDocs version : 1.6.1
Build Date UTC : 2025-05-14 00:04:43.869707+00:00
-->
